test/e2e/upgradetest/upgrade_test.go

```diff
@@ -114,3 +121,70 @@ func TestHealOneMemberForOldCluster(t *testing.T) {
 		t.Fatalf("failed to heal one member: %v", err)
 	}
 }
+
+// TestBackupForOldCluster tests that new backup sidecar could make backup from old cluster.
+func TestBackupForOldCluster(t *testing.T) {
+	t.Run("PV backup for old cluster", func(t *testing.T) {
+		testBackupForOldClusterWithBackupPolicy(t, e2eutil.NewPVBackupPolicy())
+	})
+	t.Run("S3 backup for old cluster", func(t *testing.T) {
+		testBackupForOldClusterWithBackupPolicy(t, e2eutil.NewS3BackupPolicy())
+	})
+}
+
+func testBackupForOldClusterWithBackupPolicy(t *testing.T, bp *spec.BackupPolicy) {
+	err := testF.CreateOperator()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		err := testF.DeleteOperator()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+	err = k8sutil.WaitEtcdTPRReady(testF.KubeCli.CoreV1().RESTClient(), 3*time.Second, 30*time.Second, testF.KubeNS)
+	if err != nil {
+		t.Fatalf("failed to see cluster TPR get created in time: %v", err)
+	}
+
+	bp.BackupIntervalInSecond = 60 * 60 * 24 // long enough that no backup was made automatically
+	bp.CleanupBackupsOnClusterDelete = true
+	cl := e2eutil.NewCluster("upgrade-test-", 3)
+	e2eutil.ClusterWithBackup(cl, bp)
+	testClus, err := e2eutil.CreateCluster(t, testF.KubeCli, testF.KubeNS, cl)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := e2eutil.DeleteCluster(t, testF.KubeCli, testClus, &e2eutil.StorageCheckerOptions{}); err != nil {
```

Comment:
Shouldn't we call `DeleteCluster()` with the correct type of StorageCheckerOptions{} based on the backup policy? So if it is type S3, we can wait for S3 storage to be deleted.
```
var storageCheckerOptions *e2eutil.StorageCheckerOptions
		switch testClus.Spec.Backup.StorageType {
		case spec.BackupStorageTypePersistentVolume, spec.BackupStorageTypeDefault:
			storageCheckerOptions = &e2eutil.StorageCheckerOptions{}
		case spec.BackupStorageTypeS3:
			storageCheckerOptions = &e2eutil.StorageCheckerOptions{
				S3Cli:    f.S3Cli,
				S3Bucket: f.S3Bucket,
			}
		}

if err := e2eutil.DeleteCluster(t, testF.KubeCli, testClus, storageCheckerOptions); err != nil {
```